lib/driver.js

```diff
@@ -781,20 +780,27 @@ class XCUITestDriver extends BaseDriver {
         this.assertFeatureEnabled(CUSTOMIZE_RESULT_BUNDLE_PATH);
       }
 
-      const startupRetries =
+      let startupRetries =
         this.opts.wdaStartupRetries ||
         (this.isRealDevice() ? WDA_REAL_DEV_STARTUP_RETRIES : WDA_SIM_STARTUP_RETRIES);
       const startupRetryInterval = this.opts.wdaStartupRetryInterval || WDA_STARTUP_RETRY_INTERVAL;
-      this.log.debug(
-        `Trying to start WebDriverAgent ${startupRetries} times with ${startupRetryInterval}ms interval`,
-      );
-      if (
-        !util.hasValue(this.opts.wdaStartupRetries) &&
-        !util.hasValue(this.opts.wdaStartupRetryInterval)
-      ) {
+
+      // These values help only xcodebuild.
+      if (this.isXcodebuildNeeded()) {
         this.log.debug(
-          `These values can be customized by changing wdaStartupRetries/wdaStartupRetryInterval capabilities`,
+          `Trying to start WebDriverAgent ${startupRetries} times with ${startupRetryInterval}ms interval`,
         );
+        if (
+          !util.hasValue(this.opts.wdaStartupRetries) &&
+          !util.hasValue(this.opts.wdaStartupRetryInterval)
+        ) {
+          this.log.debug(
+            `These values can be customized by changing wdaStartupRetries/wdaStartupRetryInterval capabilities`,
+          );
+        }
+      } else {
+        // The startup retry will be one time if the session does not need WDA build
+        startupRetries = 1;
```

Comment:
we probably should mention in docs for these caps that they only make sense if xcodebuild is used to compile/start WDA